scrapy/core/downloader/contextfactory.py

```diff
@@ -28,9 +29,17 @@ class ScrapyClientContextFactory(BrowserLikePolicyForHTTPS):
          understand the SSLv3, TLSv1, TLSv1.1 and TLSv1.2 protocols.'
         """
 
-        def __init__(self, method=SSL.SSLv23_METHOD, *args, **kwargs):
+        def __init__(self, method=SSL.SSLv23_METHOD, settings=None, *args, **kwargs):
```

Comment:
I think a common practice is to pass tls_verbose_logging, not Settings instance, and extract option value in from_settings / from_crawler.

Reply:
I'm fine with changing this, though I don't like that we need to list all optional args in the documentation and the error message, as I was going to pass yet another setting here in a different PR, maybe we can rephrase the messages.